Automattic/jetpack
# Refactor beta blocks registration

See https://github.com/Automattic/jetpack/issues/32602

## Proposed changes:
Registers the beta blocks (_AI Chat_, _Amazon_, _Blogroll_, _Google Docs Embed_, _Recipe_, _Create with Voice_) by specifying the path to their block.json file. This was made possible by https://github.com/Automattic/jetpack/pull/32697. The goal is to avoid the duplication of block metadata and use block.json as the source of truth.

In more details:
- Update `registerJetpackBlockFromMetadata` to redeclare variations icons on the client.
- Add helper to get a block's name and feature name.
- For each block (when needed):
  - Write reading direction-agnostic CSS
  - Remove the `editorScript` property from `block.json` (automatically added during the build)
  - Convert icon from dashicon to SVG in the `icon` property of `block.json`
  - Remove `icon.js`
  - Import icon helpers from the `shared-extension-utils` package
  - Register the block client and server sides with `block.json`
  - Move data from `index.js` to `block.json` and `editor.js` and delete the file
  - Move data from `attributes.js` to `block.json` and delete the file
  - Fix JS tests

### Other information:


## Jetpack product discussion


pedMtX-RS-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Enable beta blocks in the Jetpack constants section if you're using a JN site, or by adding `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );` to your `wp-config.php` otherwise.
- Test that the blocks behave as in `trunk` (some newer ones, such as _Blogroll_, might not be functional yet). This PR shouldn't change anything from a user's standpoint.

A couple of notes:
- Blocks metadata don't seem to be localized in production.
- _Blogroll_ seems to be often updated  at the moment. The icons, for instance, don't currently match. I'll make sure to fix that in a subsequent PR.
- I've removed the links from the _Recipe_ description. It didn't lead to anything and I couldn't find a relevant page to link to.

<img width="676" alt="Screenshot 2023-09-11 at 1 15 21 PM" src="https://github.com/Automattic/jetpack/assets/1620183/0b81951e-db9b-41cd-9bb7-411e9794aede">
